crates/ruff_linter/src/settings/types.rs

```diff
@@ -34,6 +34,7 @@ pub enum PythonVersion {
     Py311,
     Py312,
     Py313,
+    Py314,
```

Comment:
It's unfortunate that adding `Py314` makes it appear in the JSON schema and clap documentation without the possibility of mentioning that `Py314` support is in preview.

I think we should log a warning or even abort if a user specifies 3.14 and hasn't `preview` enabled.